### PR TITLE
Main Thread Fixes for Debugging + a small touch

### PR DESCRIPTION
**Pull request by Auros**
This PR pushes some things onto the main thread that SHOULD be on the main thread. This way, debugging using the unity profile will work. Also, I made it so the word "song" is singular if there is only 1 song installed (cause I only have one song installed and it was bothering me).

### CODE REVIEW

**Review by Auros — COMMENTED** [review]
therad

File: Loader.cs [hunk 1]
```diff
@@ -83,8 +83,15 @@ public static void OnLoad()
...
         }
 
+        private string _customWIPPath;
+        private string _customLevelsPath;
+
         private void Awake()
         {
+            _customWIPPath = Path.Combine(Application.dataPath, "CustomWIPLevels");
+            _customLevelsPath = Path.GetFullPath(CustomLevelPathHelper.customLevelsDirectoryPath);
+            _ = BeatmapLevelsModelSO;
```
Comment (ErisApps) [inline Loader.cs:93]: Assigning only to be discarded again seems unnecessary to me?

Comment (Auros) [reply Loader.cs:93]: Oh yeah, I was experimenting with needing to caching that. The property `get` does a Resources.FindObjectsOfTypeAll call so i didn't want that potentially running on another thread, hence the discard to cache the backing field.

Comment (Auros) [reply Loader.cs:93]: image

Yep, it's necessary.

Comment (ErisApps) [reply Loader.cs:93]: Ah, that wasn't clear to me. Might wanna add a tiny comment to it to clarify that it is needed. 🙂

**Review by ErisApps — APPROVED** [review]

**Comment by Meivyn** [comment]
Seems like there is still this missing. 

I suppose that it's complaining about this line

https://github.com/Kylemc1413/SongCore/blob/4e267edd44bbb7aa5639ca9cc1992ae45f31e7ae/Loader.cs#L983

But I don't know how to address that myself. So just reporting.

image

**Pull request merged**